src/material/schematics/ng-generate/mdc-migration/rules/theming-styles.ts

```diff
@@ -46,10 +46,20 @@ export class ThemingStylesMigration extends Migration<StyleMigrator[], Schematic
   }
 
   ruleHandler(rule: postcss.Rule) {
+    let isLegacySelector;
+    let isDeprecatedSelector;
+
     const migrator = this.upgradeData.find(m => {
-      return m.isLegacySelector(rule);
+      isLegacySelector = m.isLegacySelector(rule);
+      isDeprecatedSelector = m.isDeprecatedSelector(rule);
+      return isLegacySelector || isDeprecatedSelector;
     });
-    migrator?.replaceLegacySelector(rule);
+
+    if (isLegacySelector) {
```

Comment:
I think there will be cases where legacy and deprecated selectors are used in the same rule

```
.mat-checkbox .mat-checkbox-label { ... }
```